### PR TITLE
fix: notification placement via app component

### LINKED ISSUE #43467: `placement` option passed to `notification` prop in `App` component does not work
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/basic-antd-5-6-4-forked-69vn6n?file=/demo.tsx)

### Steps to reproduce

Create an App Component and pass a notification config to it via the notification prop. This config should have a placement set to "bottomLeft" (or any value other than "topRight"). The notification popup appears on the topRight.

### What is expected?

The notification popup should be displayed at a position corresponding to the placement value passed to the notification prop.

### What is actually happening?

The notification popup always appears at the topRight.

| Environment | Info |
| --- | --- |
| antd | 5.6.4 |
| React | 18.2.0 (react) |
| System | Mac OS Ventura 13.4.1 |
| Browser | Brave iVersion 1.52.130 Chromium: 114.0.5735.198 (Official Build) (x86_64) |

### PR DESCRIPTION
**Pull request by Rajil1213**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
Closes [43467](https://github.com/ant-design/ant-design/issues/43467)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix notification placement not being respected  when passed via App component        |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a918b74</samp>

This pull request improves the notification feature of the `App` component by adding a new test case, refactoring the `useNotification` file, and supporting custom notification config props. It also fixes some code style and consistency issues in the notification components and functions.

### <samp>🤖 Generated by Copilot at a918b74</samp>

*  Import `NotificationConfig` type and add test case for `App` component to check notification placement config from props ([link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8R8), [link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8R127-R166))
*  Reorder import statements in `useNotification` file to follow convention ([link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfL1-R9))
*  Define `DEFAULT_PLACEMENT` constant and use it as fallback value for notification placement ([link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfR22), [link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfL49-R57), [link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfL128-R136))
*  Import `getCloseIcon` and `PureContent` components from `PurePanel` file and use them in `useNotification` file to avoid duplication ([link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfR16))
*  Add `placement` prop to `Holder` component and pass it to `useRcNotification` hook to create notification container element ([link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfR44))
*  Modify `getStyle` function in `Holder` component to use placement value from props, global config, or default value ([link](https://github.com/ant-design/ant-design/pull/43522/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfL49-R57))

### CODE REVIEW

File: components/app/__tests__/index.test.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import rtlTest from '../../../tests/shared/rtlTest';
 import { render, waitFakeTimer } from '../../../tests/utils';
 import type { AppConfig } from '../context';
 import { AppConfigContext } from '../context';
+import { NotificationConfig } from 'antd/es/notification/interface';
```
[the remark this answers is not in the document]
Comment (Rajil1213) [reply components/app/__tests__/index.test.tsx]: The type import has been added (and sorted)

File: components/notification/useNotification.tsx [hunk 2]
```diff
@@ -1,22 +1,25 @@
+import * as React from 'react';
+
```
Comment (zombieJ) [inline components/notification/useNotification.tsx:2]: no need this empty line~

File: components/notification/useNotification.tsx [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
 import classNames from 'classnames';
 import { useNotification as useRcNotification } from 'rc-notification';
 import type { NotificationAPI } from 'rc-notification/lib';
-import * as React from 'react';
+
```
Comment (zombieJ) [inline components/notification/useNotification.tsx:6]: no need this empty line~

File: components/notification/useNotification.tsx (hunk 2 above)
Comment (Rajil1213) [reply components/notification/useNotification.tsx]: The extra newline has been removed.

File: components/notification/useNotification.tsx (hunk 3 above)
Comment (Rajil1213) [reply components/notification/useNotification.tsx]: The extra newline has been removed.

**Review by zombieJ — APPROVED** [review]

**Pull request merged**